File: 905-sort-array-by-parity/905-sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        int i = 0, j = A.size() - 1;
        while (i < j) {
            if (A[i] % 2 > A[j] % 2) {
                int tmp = A[i];
                A[i] = A[j];
                A[j] = tmp;
            }

            if (A[i] % 2 == 0) i++;
            if (A[j] % 2 == 1) j--;
        }

        return A;
    }
};

File: 905-sort-array-by-parity/905-sort-array-by-parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function(nums) {
    let arr = [];
    const n = arr.length;
    arr = nums.filter( (x) => x%2 == 0 ).concat(nums.filter( (x) => x%2 != 0));
    return arr;
};
